screens/DoubleSearch.js

```diff
@@ -42,27 +42,54 @@ function DoubleSearch(props) {
     const [coordinatesFrom, setCoordinatesFrom] = React.useState(null);
     const [coordinatesTo, setCoordinatesTo] = React.useState("");
     const [currentLocationCoords, setCurrentLocationCoords] = React.useState(null);
+    const [pointOfInterest, setPointOfInterest] = React.useState(null);
+    const [coordinatesPOI, setCoordinatesPOI] = React.useState({ latitude: null, longitude: null });
+
+
+    const namePointOfInterest = props.navigation.getParam("name_POI", null);
+    const latitudePointOfInterest = props.navigation.getParam("latitude_POI", null);
+    const longitudePointOfInterest = props.navigation.getParam("longitude_POI", null);
+    
+    /**
+     * Adding the point of interest as an item in the dropdownlist of the Destination search bar
+     */
+    const addItem = () => {
+            destinationItems.unshift({ id: 0, name: namePointOfInterest});   
+    }
+    const [courseScheduleLocation, setCourseScheduleLocation] = React.useState("");
+    const [initialTo, setInitialTo] = React.useState("");
+    const [placeholder, setPlaceholder] = React.useState("");
+
+
     /**
-     * Description: Method to go back to the previous screen.
+     * Description: Method to back to the previous screen.
      * Using Stack navigator.
      */
 
     // var fromScreen; 
-    const CourseScheduleDetailsScreen = props.navigation.getParam("CourseScheduleDetailsScreen", "null");
+    const CourseSchedule = props.navigation.getParam("CourseScheduleScreen", "null");
+    const CourseScheduleLocation = props.navigation.getParam("CourseScheduleLocation", "");
     const NearbyInterestDetailsScreen = props.navigation.getParam("NearbyInterestDetailsScreen", "null");
+    const InitialTo = props.navigation.getParam("destinationName", "");
+
+
     const goBack = () => {
```

Comment:
## Idea 💡 
Instead of having mixed logic like that because we want to retrieve from which context we are coming from, why we simply don't add _DoubleSearch, PreviewDirections, PreferenceMenu and Directions_ in the _**PointOfInterest**_ drawer? It is not really code duplication since we are rewritting anything, but changing the name of the screen to have a distinction (developer wise) but, it would fix all the issues of goBack()!!!!!!

Reply:
Interesting point! However, Double Search, PreviewDirections, PreferenceMenu, and Directions would have to be added to CourseScheduleStack as well since they follow the same logic in term of path

Reply:
Yess I point that out later on

Reply:
Done!